Lib/idlelib/autocomplete_w.py

```diff
@@ -242,37 +242,46 @@ def winconfig_event(self, event):
         self.is_configuring = True
         if not self.is_active():
             return
-        # Position the completion list window
-        text = self.widget
-        text.see(self.startindex)
-        x, y, cx, cy = text.bbox(self.startindex)
-        acw = self.autocompletewindow
-        if platform.system().startswith('Windows'):
-            # On Windows an update() call is needed for the completion list
-            # window to be created, so that we can fetch its width and
-            # height. However, this is not needed on other platforms (tested
-            # on Ubuntu and macOS) but at one point began causing freezes on
-            # macOS.  See issues 37849 and 41611.
-            acw.update()
-        acw_width, acw_height = acw.winfo_width(), acw.winfo_height()
-        text_width, text_height = text.winfo_width(), text.winfo_height()
-        new_x = text.winfo_rootx() + min(x, max(0, text_width - acw_width))
-        new_y = text.winfo_rooty() + y
-        if (text_height - (y + cy) >= acw_height # enough height below
-            or y < acw_height): # not enough height above
-            # place acw below current line
-            new_y += cy
-        else:
-            # place acw above current line
-            new_y -= acw_height
-        acw.wm_geometry("+%d+%d" % (new_x, new_y))
-        acw.update_idletasks()
+
+        # Since the <Configure> event may occur after the completion window is gone,
+        # catch potential TclError exceptions when accessing acw.  See: bpo-41611.
+        try:
+            # Position the completion list window
+            text = self.widget
+            text.see(self.startindex)
```

Comment:
Heavy-handed?  We could move the 'try' down a few lines.   I considered whether 'see' is useful even when code below fails, but cannot see that it is.  So I am OK with enclosing the entire logical block.

If we caught TclError after 261, what would we do?  To return, we would have to duplicate the cleanup code after the new pass.  To 'break', we would have to wrap this block with a fake loop.  Both are worse.

Reply:
Well I did write "*somewhat* heavy-handed" ;)

But yes, I too believe this is actually a good solution, besides perhaps having a more consistent mechanism to avoid uncaught exceptions in event handlers.